commit 23f8d964f15a21991a166f0d1a7cf3cf8e4bfc9b
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Fri Jul 17 16:55:00 2020 -0500

    ASoC: Intel: Skylake: Avoid the use of one-element array
    
    One-element arrays are being deprecated[1]. Replace the one-element
    array with a simple value type 'u8 reserved'[2], once it seems this
    is just a placeholder for alignment.
    
    [1] https://github.com/KSPP/linux/issues/79
    [2] https://github.com/KSPP/linux/issues/86
    
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Tested-by: kernel test robot <lkp@intel.com>
    Reviewed-by: Amadeusz Sławiński <amadeuszx.slawinski@linux.intel.com>
    Link: https://github.com/GustavoARSilva/linux-hardening/blob/master/cii/0-day/skylake-20200717.md
    Link: https://lore.kernel.org/r/20200717215500.GA13910@embeddedor
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index 9889f728752c..5e93ad85e06d 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -97,7 +97,7 @@ struct skl_audio_data_format {
 	u8 number_of_channels;
 	u8 valid_bit_depth;
 	u8 sample_type;
-	u8 reserved[1];
+	u8 reserved;
 } __packed;
 
 struct skl_base_cfg {